rust/src/nfs/nfs.rs

```diff
@@ -1033,81 +1100,31 @@ impl NFSState {
-                            SCLogDebug!("large record {}, likely file xfer", rec_size);
-
-                            // quick peek, are in WRITE mode?
-                            if rpc_phdr.procedure == NFSPROC3_WRITE {
-                                SCLogDebug!("CONFIRMED WRITE: large record {}, file chunk xfer", rec_size);
-
-                                // lets try to parse the RPC record. Might fail with Incomplete.
-                                match parse_rpc(cur_i) {
-                                    Ok((remaining, ref rpc_record)) => {
-                                        match parse_nfs3_request_write(rpc_record.prog_data) {
-                                            Ok((_, ref nfs_request_write)) => {
-                                                // deal with the partial nfs write data
-                                                self.process_partial_write_request_record(rpc_record, nfs_request_write);
-                                                cur_i = remaining; // progress input past parsed record
-                                            },
-                                            _ => {
-                                                self.set_event(NFSEvent::MalformedData);
-                                            },
-                                        }
-                                    },
-                                    Err(Err::Incomplete(_)) => {
-                                        // we just size checked for the minimal record size above,
-                                        // so if options are used (creds/verifier), we can still
-                                        // have Incomplete data. Fall through to the buffer code
-                                        // and try again on our next iteration.
-                                        SCLogDebug!("TS data incomplete");
-                                        // fall through to the incomplete below
-                                    },
-                                    Err(Err::Error(_e)) |
-                                    Err(Err::Failure(_e)) => {
-                                        self.set_event(NFSEvent::MalformedData);
-                                        SCLogDebug!("Parsing failed: {:?}", _e);
-                                        return AppLayerResult::err();
-                                    },
-                                }
-                            }
-                        }
-                        // make sure we pass a value higher than current input
-                        // but lower than the record size
-                        let n1 = cmp::max(cur_i.len(), 1024);
-                        let n2 = cmp::min(n1, rec_size);
-                        return AppLayerResult::incomplete((i.len() - cur_i.len()) as u32, n2 as u32);
+                        return self.parse_tcp_partial_data_ts(i, cur_i, rpc_phdr, rec_size);
```

Comment:
Why do not we just buffer TCP with `AppLayerResult::incomplete` ?

Reply:
the parser can handle the large records w/o buffering. Records can be several megs, which would require much buffering

Reply:
also means data is available to the detect engine sooner

Reply:
> Records can be several megs

That is the point I missed, thanks
Do you have such a pcap ?

Reply:
only in tests based on private pcaps, sadly